processing/src/main/java/org/apache/druid/query/OverrideDefaultQueryContext.java

```diff
@@ -19,43 +19,48 @@
 
 package org.apache.druid.query;
 
-import com.fasterxml.jackson.annotation.JsonProperty;
-import org.apache.druid.query.QueryContexts.Vectorize;
-import org.apache.druid.segment.QueryableIndexStorageAdapter;
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * A user configuration holder for all query types.
  * Any query-specific configurations should go to their own configuration.
- *
  * @see org.apache.druid.query.groupby.GroupByQueryConfig
  * @see org.apache.druid.query.search.SearchQueryConfig
  * @see org.apache.druid.query.topn.TopNQueryConfig
  * @see org.apache.druid.query.metadata.SegmentMetadataQueryConfig
  * @see org.apache.druid.query.scan.ScanQueryConfig
+ *
+ * This class config map is populated by any runtime property prefixed with druid.query.override.default.context
+ * Note that config values should not be directly retrieved from this class but instead should
+ * be read through {@link QueryContexts}. This class contains configs from runtime property which is then merged with
+ * configs passed in query context. The result of the merge is subsequently stored in the query context.
+ * The order of precedence in merging of the configs is as follow:
+ * runtime property values (store in this class) override by query context parameter passed in with the query
+ *
+
  */
-public class QueryConfig
+public class OverrideDefaultQueryContext
 {
-  @JsonProperty
-  private Vectorize vectorize = QueryContexts.DEFAULT_VECTORIZE;
-
-  @JsonProperty
-  private int vectorSize = QueryableIndexStorageAdapter.DEFAULT_VECTOR_SIZE;
+  private Map<String, Object> configs = new HashMap<>();
 
-  public Vectorize getVectorize()
+  @JsonAnyGetter
```

Comment:
This doesn't seem very flexible or future proof, what if we ever want to add any defaults that are not context parameters? The way this currently is we would need to add another config class that either has this as a property and remove the binding to this, or we ensure that doesn't itself have a `context` property and just have bindings for both. Why not just making this `@JsonProperty("context")` and binding the class itself to the parent path? Is there some benefit to doing it this way that I'm missing?

Reply:
Done.